File: storage/src/androidTest/java/pl/expensive/storage/DatabaseSchemaTest.java
package pl.expensive.storage;

import android.support.test.runner.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.List;

import static com.google.common.truth.Truth.assertThat;
import static pl.expensive.storage.DatabaseSchemaTestHelper.getTableColumns;

@RunWith(AndroidJUnit4.class)
public class DatabaseSchemaTest {
    private Database database = Injector.provideDatabase();

    @Test
    public void columnsForWalletTable() {
        List<String> columns = getTableColumns(database.getReadableDatabase(), "tbl_wallet");

        assertThat(columns)
                .containsExactly("uuid", "name");
    }

    @Test
    public void columnsForTransactionTable() {
        List<String> columns = getTableColumns(database.getReadableDatabase(), "tbl_transaction");

        assertThat(columns)
                .containsExactly("uuid", "amount", "currency", "date", "description", "wallet_uuid");
    }
}

File: storage/src/androidTest/java/pl/expensive/storage/DatabaseSchemaTest.kt
package pl.expensive.storage

import android.support.test.runner.AndroidJUnit4
import com.google.common.truth.Truth.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import pl.expensive.storage.DatabaseSchemaTestHelper.getTableColumns

@RunWith(AndroidJUnit4::class)
class DatabaseSchemaTest {
    private val database = Injector.provideDatabase()

    @Test
    fun columnsForTransactionTable() {
        val columns = getTableColumns(database.readableDatabase, "tbl_transaction")

        assertThat(columns)
                .containsExactly("uuid", "amount", "currency", "date", "description")
    }

    @Test
    fun columnsForCurrencyTable() {
        val columns = getTableColumns(database.readableDatabase, tbl_currency)

        assertThat(columns)
                .containsExactly("code", "format")
    }
}
